rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -119,6 +119,16 @@ def ensure_global(name, default):
     # TODO(aprotyas): Migrate files under the `include` tag in the project's Doxygen
     # configuration into the Sphinx project tree used to run the Sphinx builder in.
     extensions.append('myst_parser')
+
+# if no `primary_domain` option provided, defaults to 'cpp'
+if rosdoc2_settings.get('primary_domain', 'cpp'):
+    print(f"[rosdoc2] setting primary domain to '{{rosdoc2_settings.get('primary_domain')}}'",
+        file=sys.stderr)
+    # Tell sphinx what the primary language being documented is.
+    primary_domain = rosdoc2_settings.get('primary_domain')
```

Comment:
What if the user already set `primary_domain` or `highlight_language`? We'd be overwriting them. I think it would be better to only set these if `primary_domain` is explicitly in `rosdoc2_settings`, or if the user didn't specify `primary_domain` or `highlight_language`.

Reply:
With the changes in ac4a5ea595ffd00a64d99c7a7eed4dc62bc773a9, the last talking point is about a reasonable default for the domain/highlighting - check https://github.com/ros-infrastructure/rosdoc2/pull/24#issuecomment-893979131. For example, if `cpp` were the default provided by `rosdoc2` in case the user didn't specify `primary_domain` or `highlight_language` in their `conf.py` configuration, the following snippet should do the trick:

```python
ensure_global('primary_domain', 'cpp')
ensure_global('highlight_language', 'cpp')

if rosdoc2_settings.get('primary_domain'):
    ...
```

Thoughts?

Reply:
Hmm, I wouldn't default to `cpp` I think. The user can always explicitly label the language of code in their doc blocks right? Or does this cover a case where that doesn't work?

If we do pick a default other than the sphinx default, I'd say we should do `cpp` if it is an `ament_cmake` package and `py` or do nothing if it is an `ament_python` package.

Reply:
> Or does this cover a case where that doesn't work?

Off the top of my head, I don't think this would cover any case where the user couldn't explicitly label the code block.

> I'd say we should do `cpp` if it is an `ament_cmake` package and `py` or do nothing if it is an `ament_python` package.

Alright, I will keep this in mind as I'm working on #19.